config/session.php

```diff
@@ -99,7 +100,7 @@
 	|
 	*/
 
-	'store' => env('SESSION_STORE'),
+	'store' => env('SESSION_DRIVER'),
```

Comment:
Are you sure that this is equivalent? I haven't look into the source code nor the documentation, but my gut feeling tells me that `SESSION_DRIVER` ans `SESSION_STORE` are different things. I would have assume, that the meaning of `SESSION_STORE` depends on the value of `SESSION_DRIVER`. For example, if `SESSION_DRIVER` equals `file` I would have expected `SESSION_STORE` to be the actual physical path, e.g. `storage/session`.

I don't know it, that is just me feeling. Have you confirmed with the Laravel source code that your change is correct?

Reply:
> They have the same names as the driver name itself, so they are equivalent.

Yes and no. Only because they accidentally happen to have the same values, does not mean they are equivalent. Actually, I would call it a fail of the Laravel example configuration to use the same values for `SESSION_DRIVER` and `SESSION_STORE` in their example file because this suggests that there would be some relation which isn't.

That is the same mistake so many math teachers do on a blackboard if they - for example - need to draw to lines or triangles: most people tend to draw either parallel or orthogonal lines and tend to draw either right-angled triangles or equilateral triangle - only rarely people draw really skew lines and oblique triangles. Then, when they prove a statement, they falsely use an "accidental" property which does not hold in full generality.

Here we have the same thing. Let's assume we wanted to store the session in a database (Lychee does not). Further assume that we wanted to use two databases, for example one database for volatile, ephemeral data and one for data which needs to be persisted. Maybe the databases are even of completely different type: the first one might be a MySQL database with in-memory tables and the second one is a PostgeSQL database on disk. In this case the store in `cache.php` would not be called `'database' `, but maybe `'fast-in-mem-mysql'` and `'durable-psql'`, but the driver would still be `'database'` because this relates to the PHP class which implements the driver.

Hence, I would prefer one out of two options:

a) Either go the full way and make the whole thing completely not configurable via environment variables.
b) Revert the change to support configuration via environment variables in full generality without making assumptions.

Maybe a) is even the best option, because I doubt that we will ever support something else than we have now.

Reply:
Yes and no.

I would not recommend removing the option for the very reasons you pointed out yourself, but I wouldn't keep it the current way either. I would revert it to the old behavior, i.e. keeping `SESSION_STORE` and `SESSION_DRIVER` distinct.

As I wrote, they are semantically different things. That all examples accidentally happen to use the same values for them is a bad co-incident, which suggests a relationship between these options and their values which does simply not exist.